Lombiq.Hosting.MediaTheme.Bridge/Services/MediaThemeStep.cs

```diff
@@ -30,7 +34,15 @@ protected override async Task HandleAsync(RecipeExecutionContext context)
 
         if (model.ClearMediaThemeFolder)
         {
-            await _mediaFileStore.TryDeleteDirectoryAsync(_mediaFileStore.Combine(Paths.MediaThemeRootFolder));
+            await _mediaFileStore.TryDeleteDirectoryAsync(Paths.MediaThemeRootFolder);
+        }
+
+        // If a remote storage implementation is used, it needs to be purged too to make sure all files are fresh. This
+        // is easiest to do by removing the whole folder even if the theme Media folder itself wasn't cleared.
+        var mediaFileStoreCache = _serviceProvider.GetService<IMediaFileStoreCache>();
+        if (mediaFileStoreCache != null)
```

Comment:
I like to inline these:

```suggestion
        if (_serviceProvider.GetService<IMediaFileStoreCache>() is { } mediaFileStoreCache)
```